### PR TITLE
client: Making the configure functions more readable

### PR DESCRIPTION
**Pull request by stlaz**
Make configure file creation functions in client more readable by using setOption/setSection methods.

### CODE REVIEW

File: ipaclient/install/client.py [hunk 1]
```diff
@@ -765,181 +641,70 @@ def configure_krb5_conf(
...
-            'type': 'option',
-            'value': cli_realm
-        },
-        {
-            'name': client_hostname,
-            'type': 'option',
-            'value': cli_realm
-        }
-    ]
+    dropts = [krbconf.setOption('.' + cli_domain, cli_realm),
```
Comment (mbasti-rh) [inline ipaclient/install/client.py]: - Why do you need to change format to '+' operator
- Why do you changed indentation? IMO it is better for long lists to have
```
l = [
    first
    second
...
    N
]
```
It keeps PEP8 when you change name of variable, and it makes diffs smaller when you add/remove entries from list definition

File: ipaclient/install/client.py [hunk 2]
```diff
@@ -765,181 +641,70 @@ def configure_krb5_conf(
...
-                'type': 'option',
-                'value': cli_realm
-            },
-            {
-                'name': client_domain,
-                'type': 'option',
-                'value': cli_realm
-            }
-        ])
+        dropts.extend([krbconf.setOption('.'+client_domain, cli_realm),
```
Comment (mbasti-rh) [inline ipaclient/install/client.py]: + operator should have spaces around to be consistent with previous one, why do you need to replace format?

File: ipaclient/install/client.py [hunk 3]
```diff
@@ -765,181 +641,70 @@ def configure_krb5_conf(
...
 
-    opts.extend([
-        {
-            'name': 'libdefaults',
-            'type': 'section',
-            'value': libopts
-        },
-        IPAChangeConf.EMPTY_LINE,
-    ])
+    opts.extend([{'name': 'libdefaults', 'type': 'section', 'value': libopts},
```
Comment (mbasti-rh) [inline ipaclient/install/client.py]: I dont think that squashing dict definition to one line makes it more readable

**Comment by mbasti-rh** [comment]
ACK

**Comment by jcholast** [comment]
Pushed to installer-refactoring.

**Pull request closed without merging**